thinkhazard_processing/processing.py

```diff
@@ -0,0 +1,203 @@
+    )
+from .models import (
+    HazardSet,
+    Layer,
+    Output,
+    )
+from . import settings
+
+
+class ProcessException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+def process(hazardset_id=None, force=False):
+    hazardsets = DBSession.query(HazardSet)
+    hazardsets = hazardsets.filter(HazardSet.complete.is_(True))
+    if hazardset_id is not None:
+        hazardsets = hazardsets.filter(HazardSet.id == hazardset_id)
+    if not force:
+        hazardsets = hazardsets.filter(HazardSet.processed.is_(False))
+    if hazardsets.count() == 0:
+        print 'No hazardsets to process'
+        return
+    for hazardset in hazardsets:
+        process_hazardset(hazardset, force=force)
+
+
+def process_hazardset(hazardset, force=False):
+    print hazardset.id
+    chrono = datetime.datetime.now()
+    last_percent = 0
+
+    level_VLO = HazardLevel.get(u'VLO')
+
+    if hazardset is None:
+        raise ProcessException('HazardSet {} does not exist.'
+                               .format(hazardset.id))
+
+    if hazardset.processed:
+        if force:
+            hazardset.processed = False
+        else:
+            raise ProcessException('HazardSet {} has already been processed.'
+                                   .format(hazardset.id))
+
+    # clean previous outputs
+    DBSession.query(Output) \
+        .filter(Output.hazardset_id == hazardset.id) \
+        .delete()
+    DBSession.flush()
+
+    hazardtype = hazardset.hazardtype
+    hazardtype_settings = settings['hazard_types'][hazardtype.mnemonic]
+    thresholds = hazardtype_settings['thresholds']
+
+    project = partial(
+        pyproj.transform,
+        pyproj.Proj(init='epsg:3857'),
+        pyproj.Proj(init='epsg:4326'))
+
+    layers = {}
+    for level in (u'HIG', u'MED', u'LOW'):
+        hazardlevel = HazardLevel.get(level)
+        layer = DBSession.query(Layer) \
+            .filter(Layer.hazardset_id == hazardset.id) \
+            .filter(Layer.hazardlevel_id == hazardlevel.id) \
+            .one()
+        layers[level] = layer
+
+    with rasterio.drivers():
+        with rasterio.open(layers['HIG'].path()) as src_hig, \
+                rasterio.open(layers['MED'].path()) as src_med, \
+                rasterio.open(layers['LOW'].path()) as src_low:
+            readers = {}
+            readers['HIG'] = src_hig
+            readers['MED'] = src_med
+            readers['LOW'] = src_low
+
+            polygon_hig = polygonFromBounds(src_hig.bounds)
+            polygon_med = polygonFromBounds(src_med.bounds)
+            polygon_low = polygonFromBounds(src_low.bounds)
+            polygon = cascaded_union((
+                polygon_hig,
+                polygon_med,
+                polygon_low))
+
+            adminlevel_REG = AdminLevelType.get(u'REG')
+
+            admindivs = DBSession.query(AdministrativeDivision) \
+                .filter(AdministrativeDivision.leveltype_id == adminlevel_REG.id) \
```

Comment:
Very minor, but we could use a join here.
```
DBSession.query(AdministrativeDivision) \
    .join(AdminLevelType) \
    .filter(AdminLevelType.mnemonic == u'REG')
```
It saves one query, shouldn't be less efficient, makes code easier to ready (less variables), and is more SQLAlchemy-like.